fix(test): `TestFindAvailablePortUnique` flaky test failure on RHEL-9 / Fedora-41

## Summary

Build and Test CI fails on **rhel-9** and **fedora-41** after merging PR #221 (`fix/path-injection-hardening`).
The failure is caused by a **pre-existing flaky test** (`TestFindAvailablePortUnique`) unrelated to the PR changes.

- **CI Run**: https://github.com/alpacax/alpamon/actions/runs/22837953024
- **Failed Jobs**: `Build and Test (rhel-9)`, `Build and Test (fedora-41)`
- **Failed Step**: `Run Tests`
- **Failed Test**: `TestFindAvailablePortUnique` in `pkg/runner/editor_test.go:24`

## Error Message

```
=== RUN   TestFindAvailablePortUnique
    editor_test.go:24:
            Error:      Should be false
            Test:       TestFindAvailablePortUnique
            Messages:   Port should be unique across calls
--- FAIL: TestFindAvailablePortUnique (0.00s)
FAIL    github.com/alpacax/alpamon/pkg/runner    0.014s
```

## Root Cause

### The Test

```go
// pkg/runner/editor_test.go:19-27
func TestFindAvailablePortUnique(t *testing.T) {
    ports := make(map[int]bool)
    for i := 0; i < 10; i++ {
        port, err := findAvailablePort()
        assert.NoError(t, err, "Failed to find available port")
        assert.False(t, ports[port], "Port should be unique across calls")  // FAILS HERE
        ports[port] = true
    }
}
```

### The Function Under Test

```go
// pkg/runner/editor.go:524-531
func findAvailablePort() (int, error) {
    listener, err := net.Listen("tcp", "127.0.0.1:0")
    if err != nil {
        return 0, fmt.Errorf("failed to find available port: %w", err)
    }
    defer func() { _ = listener.Close() }()
    return listener.Addr().(*net.TCPAddr).Port, nil
}
```

### Why It Fails

1. `findAvailablePort()` binds to `127.0.0.1:0` → OS assigns an ephemeral port
2. The function records the port number and **immediately closes the listener** (defer)
3. The port returns to the OS's available pool
4. On the next call, the OS **may reassign the same port** — this is normal OS behavior

The test assumes 10 consecutive calls will always return unique ports, but this is **not guaranteed** by the OS.

### Why Only RHEL-9 and Fedora-41

| Distro | Kernel | Port Reuse Behavior |
|--------|--------|-------------------|
| RHEL-9 | 5.14+ | More aggressive ephemeral port recycling |
| Fedora-41 | 6.x | Latest kernel with updated port allocation |
| Ubuntu/Debian | 5.15-6.x | Sequential allocation happens to avoid collision in 10 calls |

All distros run as Docker containers on `ubuntu-latest` GitHub Actions runners. The container's loopback interface port allocation is governed by the host kernel's networking stack. Newer kernels (RHEL-9, Fedora-41 base images) may configure `net.ipv4.ip_local_port_range` or `SO_REUSEADDR` differently, making port reuse more likely in rapid succession.

> **Important**: Other distros passing is coincidental, not guaranteed. This test is flaky by design.

### Relationship to PR #221

**None**. The files `editor_test.go` and `editor.go` were **not modified** in PR #221. The PR only changed:
- `pkg/runner/ftp.go` — path validation
- `pkg/runner/pty.go` — WebSocket URL validation
- `pkg/runner/pty_linux.go` / `pty_darwin.go` — UID/GID range checks
- `pkg/utils/fs.go` — empty-path guards
- `pkg/utils/utils.go` — `ParseUint` for 32-bit portability

The flaky test was introduced in commit `64d76f6` and coincidentally surfaced during this CI run.

## Proposed Fix

### Option 1: Remove uniqueness assertion (Recommended)

The real usage of `findAvailablePort()` (in `CodeServerManager.doStart`) calls it **once**. Port uniqueness across multiple calls is not a requirement. The test should verify the function returns a valid port, not that 10 sequential calls never collide.

```go
func TestFindAvailablePortUnique(t *testing.T) {
    for i := 0; i < 10; i++ {
        port, err := findAvailablePort()
        assert.NoError(t, err, "Failed to find available port")
        assert.True(t, port > 0 && port <= 65535, "Port should be in valid range")
    }
}
```

### Option 2: Hold listeners open during test

If uniqueness verification is desired, keep listeners open so the OS cannot reassign ports:

```go
func TestFindAvailablePortUnique(t *testing.T) {
    var listeners []net.Listener
    defer func() {
        for _, l := range listeners {
            _ = l.Close()
        }
    }()

    ports := make(map[int]bool)
    for i := 0; i < 10; i++ {
        l, err := net.Listen("tcp", "127.0.0.1:0")
        assert.NoError(t, err)
        listeners = append(listeners, l)
        port := l.Addr().(*net.TCPAddr).Port
        assert.False(t, ports[port], "Port should be unique while listeners are held open")
        ports[port] = true
    }
}
```

> Note: Option 2 tests OS behavior, not the `findAvailablePort()` function itself. Option 1 is preferred.

## Files to Modify

- `pkg/runner/editor_test.go` — Fix `TestFindAvailablePortUnique`

## Verification

```bash
# Run the specific test
go test -v -run TestFindAvailablePortUnique ./pkg/runner/

# Run full test suite
go test -v ./... -p 1
```

After fix, re-run CI to confirm all distros pass including rhel-9 and fedora-41.

--- a/pkg/runner/editor_test.go
+++ b/pkg/runner/editor_test.go
@@ -17,12 +17,10 @@ func TestFindAvailablePort(t *testing.T) {
 }
 
 func TestFindAvailablePortUnique(t *testing.T) {
-	ports := make(map[int]bool)
 	for i := 0; i < 10; i++ {
 		port, err := findAvailablePort()
 		assert.NoError(t, err, "Failed to find available port")
-		assert.False(t, ports[port], "Port should be unique across calls")
-		ports[port] = true
+		assert.True(t, port > 0 && port <= 65535, "Port should be in valid range")
 	}
 }
 

--- a/pkg/runner/editor_test.go
+++ b/pkg/runner/editor_test.go
@@ -16,7 +16,7 @@ func TestFindAvailablePort(t *testing.T) {
 	assert.True(t, port <= 65535, "Port should be less than or equal to 65535")
 }
 
-func TestFindAvailablePortUnique(t *testing.T) {
+func TestFindAvailablePortRepeated(t *testing.T) {
 	for i := 0; i < 10; i++ {
 		port, err := findAvailablePort()
 		assert.NoError(t, err, "Failed to find available port")

